controllers/volumesnapshotbackup_controller.go

```diff
@@ -164,24 +142,20 @@ func (r *VolumeSnapshotBackupReconciler) Reconcile(ctx context.Context, req ctrl
 			})
 	}
 
-	// get VSB again before updating status here
-	// since it has been updated in reconcile batch, resourceVersion has changed
-	// prevents "the object has been modified; please apply your changes to the latest version and try again" err
-	vsb = volsnapmoverv1alpha1.VolumeSnapshotBackup{}
-	if err := r.Client.Get(ctx, req.NamespacedName, &vsb); err != nil {
-		// ignore is not found error
-		if k8serrors.IsNotFound(err) {
-			return result, nil
-		}
-		r.Log.Error(err, "unable to fetch VolumeSnapshotBackup CR")
-		return result, err
-	}
-
 	statusErr := r.Client.Status().Update(ctx, &vsb)
 	if err == nil { // Don't mask previous error
 		err = statusErr
 	}
 
+	VSBComplete, err := r.setVSBRepSourceStatus(r.Log)
```

Comment:
Shouldnt we handle err != nil condition here ?

Reply:
In this case we are handling the err by checking for `!VSBComplete`

Reply:
So you're saying err is useless. So then why is it even returned?

Reply:
Not useless. If there is an error it is also false. However sometimes it is false without being an err.